src/ThreeJsPanel.ts

```diff
@@ -308,6 +311,55 @@ export class ThreeJsPanel {
     this.axisCamera.position.set(-this.axisOffset[0], -this.axisOffset[1], this.axisScale * 2.0);
   }
 
+  orthoScreenPixelsToPhysicalUnits(pixels: number, scale: number): number {
+    // at orthoScale = 0.5, the viewport is 1 world unit tall
+    const worldUnitsPerPixel = this.orthoScale * 2 / this.getHeight();
+    return pixels * window.devicePixelRatio * worldUnitsPerPixel * scale;
```

Comment:
I would just add a comment explaining that this.getHeight already has the devicePixelRatio multiplied in because it's coming from the threejs renderer which sets things up that way by default.